scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,410 @@
+    specDict: A dictionary containing specifications for the different properties of the statistical variable.
+    columnList: A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+
+  Returns:
+    A dictionary mapping each column to their respective stat_var node definitions.
+    Example: {
+      "Total Civilian population": {
+        "populationType": "Person",
+        "statType": "measuredValue",
+        "measuredProperty": "Count Person"
+        "armedForceStatus": "Civilian"
+      },
+      "<column-name-2>": {}, .....,
+    }
+  """
+    col_map_obj = GenerateColMapBase(spec_dict=spec_dict,
+                                     column_list=column_list,
+                                     delimiter=delimiter)
+    return col_map_obj._generate_stat_vars_from_spec()
+
+
+class GenerateColMapBase:
+    """module to generate a column map given a list of columns of the dataset and a JSON Spec
+
+  Attributes:
+    specDict: A dictionary containing specifications for the different properties of the statistical variable.
+    columnList: A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+    delimiter: The delimiting string that is used for tokenising the column name
+  """
+
+    # TODO - add typing hints and document the format.
+    def __init__(self, spec_dict={}, column_list=[], delimiter='!!'):
+        """module init"""
+        self.features = spec_dict
+        self.column_list = column_list
+        self.column_map = {}
+        self.delimiter = delimiter
+        # TODO: Add a dictionary of counters with summary stats of number of columns, number of statvars generated, statvars of differnt population types, errors or warnings,
+        # fill missing keys in JSON spec with empty values
+        for key in _JSON_KEYS:
+            if key not in self.features:
+                # TODO: initialize UniversePVs as empty list
+                if key == 'ignoreColumns':
+                    self.features[key] = []
+                else:
+                    self.features[key] = {}
+
+    def _find_and_replace_column_names(self, column):
+        """if spec has find_and_replace defined, this function updates column names"""
+        if 'find_and_replace' in self.features['preprocess']:
+            find_and_replace_dict = self.features['preprocess'][
+                'find_and_replace']
+            # replace entire column name
+            if column in find_and_replace_dict:
+                return find_and_replace_dict[column]
+            # replace a token in the column name
+            else:
+                part_list = column.split(self.delimiter)
+                for idx, part in enumerate(part_list):
+                    if part in find_and_replace_dict:
+                        part_list[idx] = find_and_replace_dict[part]
+                        return self.delimiter.join(part_list)
```

Comment:
May be useful to support more than one token being replaced.
Can add a TODO for a future change.

Reply:
Done.